File: training-data/java/InsertionSort.java
import java.util.Arrays;

public class InsertionSort {
    public static int[] sort(int[] items) {
        int[] arr = items.clone();
        for (int i = 1; i < arr.length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
        return arr;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(sort(new int[] {5, 3, 8, 1, 9, 2})));
    }
}

File: training-data/r/insertion_sort.R
insertion_sort <- function(items) {
  for (i in 2:length(items)) {
    key <- items[i]
    j <- i - 1
    while (j >= 1 && items[j] > key) {
      items[j + 1] <- items[j]
      j <- j - 1
    }
    items[j + 1] <- key
  }
  items
}

print(insertion_sort(c(12, 11, 13, 5, 6)))
